src/js/select2/data/select.js

```diff
@@ -278,6 +278,10 @@ define([
       item._resultId = this.generateResultId(this.container, item);
     }
 
+    if (item.children) {
+        item.children = item.children.map(this._normalizeItem);
```

Comment:
Would it make sense to limit the recursion depth here?

Reply:
It's probably overly-pedantic, but it does seem that [HTML `optgroup` elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup) don't support nesting, so there could be a small safety improvement by preventing recursive calls here. 

It's unlikely that circular references or multiple levels of nesting would appear in input data often on the web, but unless there's a downside to preventing recursive calls (other than slightly less-clean code), I figure it could make sense to limit the recursion to a single level.